enhancements/kube-apiserver/audit-policy.md

```diff
@@ -261,61 +284,302 @@ the respective file-based audit policy of
-these policies are extended by allowing logging of `oauthaccesstokens` and `oauthauthorizetokens`.
+
+- `Default`:
+
+  ```yaml
+    # Log the full Identity API resource object so that the audit trail
+    # allows us to match the username with the IDP identity.
+    - level: RequestResponse
+      verbs: ["create", "update", "patch", "delete"]
+      resources:
+      - group: "user.openshift.io"
+        resources: ["identities"]
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+    # A catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - RequestReceived
+  ```
+
+- `WriteRequestBodies`:
+
+  ```yaml
+    # Log the full Identity API resource object so that the audit trail
+    # allows us to match the username with the IDP identity.
+    - level: RequestResponse
+      verbs: ["create", "update", "patch", "delete"]
+      resources:
+      - group: "user.openshift.io"
+        resources: ["identities"]
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+    # exclude resources where the body is security-sensitive
+    - level: Metadata
+      resources:
+      - group: "route.openshift.io"
+        resources: ["routes"]
+      - resources: ["secrets"]
+    - level: Metadata
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthclients"]
+    # log request and response payloads for all write requests
+    - level: RequestResponse
+      verbs:
+      - update
+      - patch
+      - create
+      - delete
+      - deletecollection
+    # catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - RequestReceived
+  ```
+
+- `AllRequestBodies`:
+
+  ```yaml
+    # exclude resources where the body is security-sensitive
+    - level: Metadata
+      resources:
+      - group: "route.openshift.io"
+        resources: ["routes"]
+      - resources: ["secrets"]
+    - level: Metadata
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthclients"]
+    # catch-all rule to log all other requests with request and response payloads
+    - level: RequestResponse
+  ```
+
+### API Details
+
+The YAML of the new API should look like this:
+
+```yaml
+kind: APIServer
+apiVersion: config.openshift.io/v1
+spec:
+  ...
+  audit:
+    profile: None | Default | WriteRequestBodies | AllRequestBodies
+    customRules:
+    - group: system:authenticated:oauth
+      profile: None |Default | WriteRequestBodies | AllRequestBodies
+```
+
```

Comment:
Why do we need the seperate `Profile` then also the list
- group
- profile
Kinda a repeat of the question here:
https://github.com/openshift/api/pull/930/files#r645170272